_pytest/python.py

```diff
@@ -49,7 +49,10 @@ def _has_positional_arg(func):
 
 
 def filter_traceback(entry):
-    return entry.path != cutdir1 and not entry.path.relto(cutdir2)
+    # ensure entry.path is always a py.path.local object
+    # see https://bitbucket.org/pytest-dev/py/issues/71
+    path = py.path.local(entry.path)
```

Comment:
maybe we should not use ``entry.path`` but rather ``entry.frame.code.raw.co_filename`` and detect "<...>" (a typical way to mark dynamic code) before entering the cut/filter logic with ``py.path.local(co_filename)``.  When we detect dynamic code we should simply not filter it (i.e. return False) because we can't tell where it originated from.